### PR TITLE
Hustle: Agents, Tags, Docs Cleanup

### PR DESCRIPTION
**Pull request by jburchard**
**Added methods**
`get_agent()`
`get_agents()`
`create_agent()`
`update_agent()`
`get_tags()`
`get_tag()`

**Other**
- Cleaned up the documentation and strings from v1
- Fixed a couple of bugs in the v1 code.

### CODE REVIEW

File: parsons/hustle/hustle.py [hunk 1]
```diff
@@ -113,6 +115,103 @@ def _error_check(self, r, raise_on_error):
...
+                The email address of the agent.
+        `Returns:`
+            dict
+        """
+
+        agent = {'name': name,
+                 'fullName': email,
+                 'phoneNumber': phone_number,
+                 'sendInvite': send_invite,
+                 'email': email}
```
Comment (eliotst) [inline parsons/hustle/hustle.py]: `group_id` is not being set here.

File: parsons/hustle/hustle.py [hunk 2]
```diff
@@ -113,6 +115,103 @@ def _error_check(self, r, raise_on_error):
...
+        Update an agent.
+
+        `Args:`
+            agent_id: str
+                The agent id.
+            name: str
+                The name of the agent.
+            full_name: str
+                The full name of the agent.
+            phone_number: str
```
Comment (eliotst) [inline parsons/hustle/hustle.py]: Looks like a copy-paste error.. this docstring describes Args for `phone_number` and `email`, but they are not valid function arguments.

File: parsons/hustle/hustle.py [hunk 3]
```diff
@@ -113,6 +115,103 @@ def _error_check(self, r, raise_on_error):
...
+                The valid phone number of the agent.
+            send_invite: boolean
+                Send an invitation to the agent.
+            email:
+                The email address of the agent.
+        `Returns:`
+            dict
+        """
+
+        agent = {'groupId': agent_id,
```
Comment (eliotst) [inline parsons/hustle/hustle.py]: It doesn't look like `groupId` is a valid parameter for the `updateAgent` endpoint (https://api.hustle.com/docs/#operation/updateAgent).

File: parsons/hustle/hustle.py [hunk 4]
```diff
@@ -256,7 +359,7 @@ def create_lead(self, group_id, phone_number, first_name, last_name=None, email=
         lead = json_format.remove_empty_keys(lead)
 
         logger.debug('Generating lead for {first_name} {last_name}.')
-        return self._request(endpoint, req_type="POST", payload=lead)
+        return self._request(f'groups/{group_id}/leads', req_type="POST", payload=lead)
```
[the remark this answers is not in the document]
Comment (jburchard) [reply parsons/hustle/hustle.py]: Added 201 as a success option.

File: parsons/hustle/hustle.py [hunk 5]
```diff
@@ -113,6 +115,103 @@ def _error_check(self, r, raise_on_error):
...
+                 'fullName': email,
+                 'phoneNumber': phone_number,
+                 'sendInvite': send_invite,
+                 'email': email}
+
+        # Remove empty args in dictionary
+        agent = json_format.remove_empty_keys(agent)
+
+        logger.info(f'Generating {full_name} agent.')
+        return self._request(f'groups/{group_id}/agents', req_type="POST", payload=agent)
```
[the remark this answers is not in the document]
Comment (jburchard) [reply parsons/hustle/hustle.py]: Added 201 as a possible success option and added a Possible To Do for passing expected success for each method.

File: parsons/hustle/hustle.py (hunk 1 above)
Comment (jburchard) [reply parsons/hustle/hustle.py]: It isn't passed in as part of the payload, but rather in the construction of the url.

See line 181.

File: parsons/hustle/hustle.py [hunk 6]
```diff
@@ -113,6 +115,103 @@ def _error_check(self, r, raise_on_error):
...
+            send_invite: boolean
+                Send an invitation to the agent.
+            email:
+                The email address of the agent.
+        `Returns:`
+            dict
+        """
+
+        agent = {'name': name,
+                 'fullName': email,
```
[the remark this answers is not in the document]
Comment (jburchard) [reply parsons/hustle/hustle.py]: Done.

File: parsons/hustle/hustle.py (hunk 3 above)
Comment (jburchard) [reply parsons/hustle/hustle.py]: Fixed.

**Comment by jburchard** [comment]
@eliotst - Ready for one more pass.

**Pull request merged**